src/pmt.c

```diff
@@ -2019,7 +2020,7 @@ int process_pmt(int filter, unsigned char *b, int len, void *opaque) {
     return 0;
 }
 
-void copy_en300468_string(uint8_t *dest, int dest_len, uint8_t *src, int len) {
+void copy_en300468_string(char *dest, int dest_len, char *src, int len) {
```

Comment:
This method should use unsigned char as arguments as chars can have values over 128 which becomes negative when assigned to charset variable.

Reply:
See line 2030.

Reply:
Okay, I'll drop that part of the revert then

Reply:
Added a new commit which should address this